xarray/tests/test_dataarray.py

```diff
@@ -1392,6 +1402,45 @@ def test_expand_dims_with_scalar_coordinate(self):
         roundtripped = actual.squeeze(['z'], drop=False)
         assert_identical(array, roundtripped)
 
+    def test_expand_dims_with_greater_dim_size(self):
+        """Python 3.6+ dicts keep insertion order, unlike Python 3.5 and
+        earlier. Therefore the following tests have to use ordered dicts to
+        pass for python 3.5 and earlier.
+        """
+        array = DataArray(np.random.randn(3, 4), dims=['x', 'dim_0'],
+                          coords={'x': np.linspace(0.0, 1.0, 3), 'z': 1.0},
+                          attrs={'key': 'entry'})
+        # For python 3.5 and earlier this has to be an ordered dict, to
+        # maintain insertion order.
+        actual = array.expand_dims(
+            OrderedDict((('y', 2), ('z', 1), ('dim_1', ['a', 'b', 'c']))))
+
+        expected_coords = OrderedDict((
+            ('y', [0, 1]), ('z', [1.0]), ('dim_1', ['a', 'b', 'c']),
+            ('x', np.linspace(0, 1, 3)), ('dim_0', range(4))))
+        expected = DataArray(array.values * np.ones([2, 1, 3, 3, 4]),
+                             coords=expected_coords,
+                             dims=list(expected_coords.keys()),
+                             attrs={'key': 'entry'}
+                             ).drop(['y', 'dim_0'])
+        assert_identical(expected, actual)
+
+        # Test with kwargs instead of passing dict to dim arg.
+        other_way = array.expand_dims(y=2, z=1, dim_1=['a', 'b', 'c'])
+        # Unfortunately, there is no way to maintain insertion order with
+        # kwargs in python 3.5 and earlier, so for now we have to ensure the
+        # dimensions of the expected result are in the same order as the actual
+        # result to allow the test to pass.
```

Comment:
As long as we have reuse the standard `either_dict_or_kwargs` utility, I think it is reasonable to simplify this test to only check inserting a single dimension in a way that is both Python 3.5 and 3.6+ compatible.